src/engine/pieces/bishop.ts

```diff
@@ -1,13 +1,78 @@
 import Piece from './piece';
 import Player from '../player';
 import Board from '../board';
+import Square from "../square";
+import King from "./king";
 
 export default class Bishop extends Piece {
     public constructor(player: Player) {
         super(player);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        let currentSquare = board.findPiece(this)
+        let availableMoves = new Array(0)
+
+        // moving top-right
+        let index = 1;
+        for (let i = currentSquare.row + 1; i < 8; i++){
```

Comment:
Daca ne uitam la partile comune ale acestor for-uri, am putea extrage ceva pentru a nu fi necesare toate?